pkg/storegateway/bucket.go

```diff
@@ -719,12 +719,23 @@ func logSeriesRequestToSpan(ctx context.Context, l log.Logger, minT, maxT int64,
 		"msg", "BucketStore.Series",
 		"request min time", time.UnixMilli(minT).UTC().Format(time.RFC3339Nano),
 		"request max time", time.UnixMilli(maxT).UTC().Format(time.RFC3339Nano),
-		"request matchers", storepb.PromMatchersToString(matchers...),
-		"request block matchers", storepb.PromMatchersToString(blockMatchers...),
+		"request matchers", promMatchersToString(matchers...),
```

Comment:
Can we use `util.MatchersStringer(matchers).String()`?

Reply:
While `util.MatchersStringer(matchers).String()` should be suitable, there is the difference that `promMatchersToString` surrounds the string with curly braces. It's probably no big deal if we stop doing that, as it's just for debug logs? Also `util.MatchersStringer(matchers).String()` is more performant due to using `strings.Builder` instead of concatenating strings.

Reply:
Even better: we can pass the stringer, since both go-kit and OpenTracing allow that as a value.
That means if the trace is not sampled and debug logging is off, we don't format them at all.